### ISSUE TITLE
Release workflow fails to push changes

### ISSUE THREAD

**Issue by hohwille** [devonfw/ide#1057]
After #979 is resolved and fixed, I reran the release workflow and it sill fails now with this error:

```
remote: error: GH006: Protected branch update failed for refs/heads/master.        
remote: error: You're not authorized to push to this branch. Visit https://docs.github.com/articles/about-protected-branches/ for more information. 2 of 2 required status checks are expected.        
To https://github.com/devonfw/ide
 ! [remote rejected] master -> master (protected branch hook declined)
```

Full log:
https://github.com/devonfw/ide/actions/runs/4102046425/jobs/7074531639

**Comment by hohwille** [comment]
So the actual release itself was build and arrived in maven central:
https://repo1.maven.org/maven2/com/devonfw/tools/ide/devonfw-ide-scripts/2023.01.001/

Unlike to what I expected also the doc artifact was released and staged automatically (I did not manually release it in OSSRH):
https://repo1.maven.org/maven2/com/devonfw/tools/ide/devonfw-ide-doc/2023.01.001/

However, we now to not have the version changes comitted and are lacking the tag for the release.
Also no github release has been crafted, what is obvious as this should happen after the push (https://github.com/devonfw/ide/blob/master/.github/workflows/release.yml)
I can try to manually fake these changes but we still have to investigate how to fix this.
Is the workflow only missing out to configure git properly or are we failing due to a lack of permissions and our branch protection?

**Comment by hohwille** [comment]
I found this one:
https://github.com/stefanzweifel/git-auto-commit-action/issues/71

So we could add `---force` to the `git push` commands and I can allow force pushes to bypass the branch protection.
As we usually do not give out push permissions but only triage permissions that should be no big deal...

**Comment by maybeec** [comment]
@hohwille, should be fixed by making use of https://github.com/devonfw-actions/java-maven-setup or copying the respective section out: https://github.com/devonfw-actions/java-maven-setup/blob/main/action.yml#L117

**Comment by hohwille** [comment]
Thanks @maybeec - even better.
we already have this:
```
        git config --global user.email ${{ secrets.BUILD_USER_EMAIL }}
        git config --global user.name ${{ secrets.BUILD_USER }}
```

So all we need is:
```
git remote set-url origin github.com/devonfw/ide.git
```

**Comment by hohwille** [comment]
Fixed. I am closing in optimistic assumption that it finally works now. Otherwise, we can reopen or create a new issue.

**Comment by hohwille** [comment]
I have to reopen.
This strange workaround (to set origin to a pseudo URL without https://) does not work:
```
Run maven_config="$(cat .mvn/maven.config)"
[master fab0810] set next version to 2023.02.002-SNAPSHOT
 1 file changed, 1 insertion(+), 1 deletion(-)
fatal: 'github.com/devonfw/ide.git' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Error: Process completed with exit code 128.
```

Either I picked this wrong from the java-maven-setup action or it just does not work (anymore).

**Comment by maybeec** [comment]
You picked it wrongly. Check again the whole action step 

`name: Setup repo for push on protected branch`

**Comment by hohwille** [comment]
> You picked it wrongly. Check again the whole action step

Thanks for pointing this out. Indeed I was too quick - already was confused what kind of magic hack github would have done to accept such origin without scheme. I hope that now I got it right.

**Comment by hohwille** [comment]
It seems I have to give up with github actions - it is still not working:
```
warning: redirecting to https://github.com/devonfw/ide.git/
remote: error: GH006: Protected branch update failed for refs/heads/master.        
remote: error: 2 of 2 required status checks are expected. You're not authorized to push to this branch. Visit https://docs.github.com/articles/about-protected-branches/ for more information.        
To https://github.com/devonfw/ide.git
 ! [remote rejected] master -> master (protected branch hook declined)
```

**Comment by hohwille** [comment]
```
$ ORIGIN="$(git config --get remote.origin.url)"

$ SED_OUT=$(echo $ORIGIN | sed -r -E -n 's@^https:\/\/(github.com.*)@\1@p')

$ echo $SED_OUT
github.com/hohwille/ide.git

$ ORIGIN="https://BUILD_USER:BUILD_USER_PASSWD@$SED_OUT"

$ echo $ORIGIN
https://BUILD_USER:BUILD_USER_PASSWD@github.com/hohwille/ide.git
```
So I am doing exactly what the `java-maven-setup` action is doing (of course I fill in proper values for BUILD_USER and BUILD_USER_PASSWORD. However those have been configured by somebody else in the devonfw org, so I do not know if they fit for devonfw-ide.
Do I have to override these secrets in ide repo?

**Comment by maybeec** [comment]
sry, forgot one final thing, which is documented in the readme: https://github.com/devonfw-actions/java-maven-setup#github-composite-action-for-very-basic-java--maven--git-basics

> To be able to finally push afterwards to the repository. You have to disable default github actions authentication header. See an example [here](https://github.com/devonfw/cobigen/blob/eb8519c4dd69b307dd1e669eeab0bfba59c1c1a1/release.sh#L171)

**Comment by hohwille** [comment]
@maybeec thanks for the hint. I adapted this. Now I get this error:
```
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/devonfw/ide.git/'
Error: Process completed with exit code 128.
```

Funny. `BUILD_USER` and `BUILD_USER_PASSWD` are secrets on devonfw ORG level that I have not created...

**Comment by hohwille** [comment]
I fixed this with the same solution that @alfeilex found for update-urls workflow. As soon as it succeeded, we can close this issue.

**Comment by hohwille** [comment]
Finally works. Hooray!
